javagen/src/main/java/com/azure/autorest/template/ClientMethodTemplate.java

```diff
@@ -803,6 +816,30 @@ protected void generateLongRunningBeginAsync(ClientMethod clientMethod, JavaType
         });
     }
 
+    /**
+     * Generate long running begin async method for protocol client
+     * @param clientMethod client method
+     * @param typeBlock type block
+     */
+    protected void generateProtocolLongRunningBeginAsync(ClientMethod clientMethod, JavaType typeBlock) {
+        String contextParam = "requestOptions != null ? requestOptions.getContext() : Context.NONE";
+        String pollingStrategy = clientMethod.getMethodPollingDetails().getPollingStrategy()
+                .replace("{httpPipeline}", clientMethod.getClientReference() + ".getHttpPipeline()")
+                .replace("{context}", contextParam)
+                .replace("{serializerAdapter}", clientMethod.getClientReference() + ".getSerializerAdapter()")
+                .replace("{intermediate-type}", clientMethod.getMethodPollingDetails().getIntermediateType().toString())
+                .replace("{final-type}", clientMethod.getMethodPollingDetails().getFinalType().toString());
```

Comment:
nit, maybe share this code in a private method as duplicate in `generateLongRunningBeginAsync` method?